tests/functional/test_typed_predictor_signature.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+from pydantic import BaseModel, Field, ValidationError, field_validator
+from typing import Annotated, Optional
+from dspy.functional.typed_predictor_signature import TypedPredictorSignature
+
+
+class PydanticInput(BaseModel):
+    command: str
+
+
+class PydanticOutput1(BaseModel):
+    @field_validator("name", mode="wrap")
+    @staticmethod
+    def validate_name(name, handler):
+        try:
+            return handler(name)
+        except ValidationError:
+            return 'INVALID_NAME'
+
+    name: Annotated[str,
+                    Field(default='NOT_FOUND', max_length=15,
```

Comment:
why use Annotated instead of assignment = Field(...)?

Reply:
From Pydantic docs - "In case you use field constraints with compound types, an error can happen in some cases. To avoid potential issues, you can use Annotated:"

See https://docs.pydantic.dev/latest/concepts/fields/#validation-alias